### PR TITLE
Update readme with new failure policy details

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Part of #655

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,8 +21,6 @@ Take a look at the [concepts](https://jobset.sigs.k8s.io/docs/concepts/) page fo
 
 - **Configurable success policies**: JobSet has [configurable success policies](https://github.com/kubernetes-sigs/jobset/blob/v0.6.0/examples/simple/success-policy.yaml) which target specific ReplicatedJobs, with operators to target `Any` or `All` of their child jobs. For example, you can configure the JobSet to be marked complete if and only if all pods that are part of the “worker” ReplicatedJob are completed. This enables users to use their compute resources more efficiently, allowing a workload to be declared successful and release the resources for the next workload more quickly.
 
-- **Configurable failure policies**: JobSet has [configurable failure policies](https://github.com/kubernetes-sigs/jobset/blob/v0.6.0/examples/simple/max-restarts.yaml) which allow the user to specify a maximum number of times the JobSet should be restarted in the event of a failure. If any job is marked failed, the entire JobSet will be recreated, allowing the workload to resume from the last checkpoint. When no failure policy is specified, if any job fails, the JobSet simply fails.
```
Comment (googs1025) [inline README.md:24]: I'm a little curious. We just have to delete the old description. Do we also need to update it to the new description?

Comment (danielvegamyhre) [reply README.md:24]: What the... something weird is going on with the CI, last night the diff was correct and now it's missing the new content... let me try to fix it

Comment (danielvegamyhre) [reply README.md:24]: This is so weird, my local commit 2577eb03c0d65fdcee4802bb8132d6903d433558 has the correct changes, and *supposedly* has been pushed to remote origin:

```
danielvm@danielvm622:~/go/src/sigs.k8s.io/jobset$ git push origin readme2
Everything up-to-date
```

Yet the diff here does not include the changes.

**Pull request closed without merging**